conans/test/integration/toolchains/gnu/test_makedeps.py

```diff
@@ -0,0 +1,213 @@
+import os.path
+import platform
+import re
+import textwrap
+
+import pytest
+
+from conans.test.assets.genconanfile import GenConanfile
+from conans.test.utils.tools import TestClient
+
+
+@pytest.mark.skipif(platform.system() not in ["Linux", "Darwin"], reason="Make")
+def test_link_lib_correct_order():
+    client = TestClient()
+    liba = GenConanfile().with_name("liba").with_version("0.1")
+    libb = GenConanfile().with_name("libb").with_version("0.1").with_require("liba/0.1")
+    libc = GenConanfile().with_name("libc").with_version("0.1").with_require("libb/0.1")
+    consumer = GenConanfile().with_require("libc/0.1")
+    client.save({"liba.py": liba, "libb.py": libb, "libc.py": libc, "consumer.py": consumer})
+    client.run("create liba.py")
+    pid_a = client.created_package_reference("liba/0.1")
+    folder_a = client.get_latest_pkg_layout(pid_a).package()
+    client.run("create libb.py")
+    pid_b = client.created_package_reference("libb/0.1")
+    folder_b = client.get_latest_pkg_layout(pid_b).package()
+    client.run("create libc.py")
+    pid_c = client.created_package_reference("libc/0.1")
+    folder_c = client.get_latest_pkg_layout(pid_c).package()
+    client.run("install consumer.py -g MakeDeps")
+    deps = client.load("conanautotoolsdeps.sh")
+    # check the libs are added in the correct order with this regex
+    assert re.search("export LDFLAGS.*{}.*{}.*{}".format(folder_c, folder_b, folder_a), deps)
+
+
+@pytest.mark.skipif(platform.system() not in ["Linux", "Darwin"], reason="Make")
+def test_cpp_info_aggregation():
+
+    profile = textwrap.dedent("""
+         [settings]
+         build_type=Release
+         arch=x86
+         os=Macos
+         compiler=gcc
+         compiler.libcxx=libstdc++11
+         compiler.version=7.1
+         compiler.cppstd=17
+    """)
+
+    dep_conanfile = textwrap.dedent("""
+
+    from conan import ConanFile
+
+    class Dep(ConanFile):
+
+        settings = "os", "arch", "compiler", "build_type"
+
+        def package_info(self):
+            self.cpp_info.includedirs = []
+            self.cpp_info.includedirs.append("path/includes/{}".format(self.name))
+            self.cpp_info.includedirs.append("other\\include\\path\\{}".format(self.name))
```

Comment:
If test not Windows, why using \\ backslash in the path?

Reply:
this test is invalid, I'll remove it. I would prefer not supporting Windows with MakeDeps for now, we have NMake already for windows.